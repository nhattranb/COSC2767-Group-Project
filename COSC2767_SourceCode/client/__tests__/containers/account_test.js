/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 3 Jan 2025
 *   Last modified: 3 Jan 2025
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Account from '../../app/containers/Account';

const mockStore = configureStore([]);

describe('Account component', () => {
  let store;

  const initialState = {
    account: {
      user: {
        firstName: 'John',
        lastName: 'Doe'
      }
    },
    resetPassword: {
      resetFormData: {},
      formErrors: {}
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders correctly with user data', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Account />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with empty user data', () => {
    const emptyState = {
      ...initialState,
      account: {
        user: {
          firstName: '',
          lastName: ''
        }
      }
    };
    store = mockStore(emptyState);

    const { asFragment } = render(
      <Provider store={store}>
        <Account />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
