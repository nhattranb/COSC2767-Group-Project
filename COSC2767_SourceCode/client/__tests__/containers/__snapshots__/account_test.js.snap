// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component renders correctly with empty user data 1`] = `
<DocumentFragment>
  <div
    class="account"
  >
    <div
      class="sub-page"
    >
      <div
        class="subpage-header"
      >
        <h3
          class="mb-0"
        >
          Account Details
        </h3>
      </div>
      <div
        class="subpage-body"
      >
        <div
          class="account-details"
        >
          <div
            class="info"
          >
            <div
              class="desc"
            >
              <p
                class="one-line-ellipsis mr-3"
              >
                <span
                  class="provider-email"
                >
                  Logged in With 
                </span>
              </p>
              <span
                class="custom-badge custom-badge-secondary"
                style="border-radius: 3px;"
              >
                Member
              </span>
            </div>
          </div>
          <form>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-6"
              >
                <div
                  class="input-box "
                >
                  <label>
                    First Name
                  </label>
                  <div
                    class="input-text-block"
                  >
                    <input
                      autocomplete="on"
                      class="input-text"
                      name="firstName"
                      placeholder="Please Enter Your First Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <span
                    class="invalid-message"
                  />
                </div>
              </div>
              <div
                class="col-12 col-md-6"
              >
                <div
                  class="input-box "
                >
                  <label>
                    Last Name
                  </label>
                  <div
                    class="input-text-block"
                  >
                    <input
                      autocomplete="on"
                      class="input-text"
                      name="lastName"
                      placeholder="Please Enter Your Last Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <span
                    class="invalid-message"
                  />
                </div>
              </div>
              <div
                class="col-12 col-md-12"
              >
                <div
                  class="input-box "
                >
                  <label>
                    Phone Number
                  </label>
                  <div
                    class="input-text-block"
                  >
                    <input
                      autocomplete="on"
                      class="input-text"
                      name="phoneNumber"
                      placeholder="Please Enter Your Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <span
                    class="invalid-message"
                  />
                </div>
              </div>
            </div>
            <hr />
            <div
              class="profile-actions"
            >
              <button
                class="input-btn custom-btn-secondary md text-only icon-left "
                style="border-radius: 3px;"
                type="submit"
              >
                <span
                  class="btn-text"
                >
                  Save changes
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Account component renders correctly with user data 1`] = `
<DocumentFragment>
  <div
    class="account"
  >
    <div
      class="sub-page"
    >
      <div
        class="subpage-header"
      >
        <h3
          class="mb-0"
        >
          Account Details
        </h3>
      </div>
      <div
        class="subpage-body"
      >
        <div
          class="account-details"
        >
          <div
            class="info"
          >
            <div
              class="desc"
            >
              <p
                class="one-line-ellipsis mr-3"
              >
                <span
                  class="provider-email"
                >
                  Logged in With 
                </span>
              </p>
              <span
                class="custom-badge custom-badge-secondary"
                style="border-radius: 3px;"
              >
                Member
              </span>
            </div>
          </div>
          <form>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-6"
              >
                <div
                  class="input-box "
                >
                  <label>
                    First Name
                  </label>
                  <div
                    class="input-text-block"
                  >
                    <input
                      autocomplete="on"
                      class="input-text"
                      name="firstName"
                      placeholder="Please Enter Your First Name"
                      type="text"
                      value="John"
                    />
                  </div>
                  <span
                    class="invalid-message"
                  />
                </div>
              </div>
              <div
                class="col-12 col-md-6"
              >
                <div
                  class="input-box "
                >
                  <label>
                    Last Name
                  </label>
                  <div
                    class="input-text-block"
                  >
                    <input
                      autocomplete="on"
                      class="input-text"
                      name="lastName"
                      placeholder="Please Enter Your Last Name"
                      type="text"
                      value="Doe"
                    />
                  </div>
                  <span
                    class="invalid-message"
                  />
                </div>
              </div>
              <div
                class="col-12 col-md-12"
              >
                <div
                  class="input-box "
                >
                  <label>
                    Phone Number
                  </label>
                  <div
                    class="input-text-block"
                  >
                    <input
                      autocomplete="on"
                      class="input-text"
                      name="phoneNumber"
                      placeholder="Please Enter Your Phone Number"
                      type="text"
                      value=""
                    />
                  </div>
                  <span
                    class="invalid-message"
                  />
                </div>
              </div>
            </div>
            <hr />
            <div
              class="profile-actions"
            >
              <button
                class="input-btn custom-btn-secondary md text-only icon-left "
                style="border-radius: 3px;"
                type="submit"
              >
                <span
                  class="btn-text"
                >
                  Save changes
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
